portscanner.py

```diff
@@ -51,6 +53,21 @@ def tcp_threader():
         port = q.get()
         tcp_scan(port)
         q.task_done()
+        
+def udp_scan(port):
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        s.sendto('invalid request', (host, port))
```

Comment:
does this work? I thought you said udp scanning was too complicated?

Reply:
With this code, we can detect unclosed ports(that doesn't mean ports are opened. Some packets may be rejected by firewall).

Reply:
UDP servers won't reply for requests that don't understand, but if the port is closed the OS will trigger an exception.